app/controllers/blocks_controller.rb

```diff
@@ -1,10 +1,12 @@
 # frozen_string_literal: true
-
+require "#{Rails.root}/lib/alerts/check_timeline_manager"
 class BlocksController < ApplicationController
   respond_to :json
 
   def destroy
+    course = Block.find(params[:id]).course
```

Comment:
With this data flow, you're doing the same `find` twice. This should be done with as few and as efficient SQL queries as possible.

One alternative would be to do it like this:
`block = Block.find(params(:id).destroy`
`course = block.course`

This destroys the block first (ie, deletes the record from the database), but keeps the object as a variable so that you can still use it to find the corresponding course.